### PR TITLE
feat(discover2) Insert into transactions store in tests

### PR DESCRIPTION
**Pull request by markstory**
When saving transaction events we should insert the event into both the event dataset *and* the transactions dataset. Doing both lets us simulate production state in our test suite which is something I need to write integration tests.

Refs SEN-1043

### CODE REVIEW

File: tests/snuba/api/endpoints/test_organization_event_details.py [hunk 1]
```diff
@@ -64,6 +64,32 @@ def test_simple(self):
...
+            "sentry-api-0-organization-event-details",
+            kwargs={
+                "organization_slug": self.project.organization.slug,
+                "project_slug": self.project.slug,
+                "event_id": event.event_id,
+            },
+        )
+        with self.feature("organizations:events-v2"):
+            response = self.client.get(url, format="json")
+        assert response.status_code == 200
```
Comment (wedamija) [inline tests/snuba/api/endpoints/test_organization_event_details.py]: Do we need to check that any specific values are in the response? Or does this just pass where it used to fail before this change?

**Review by wedamija — APPROVED** [review]
Not sure if it's because of the timeout, but this made the `Backend [Postgres] (1/2)` much slower, from ~10min -> 15min. Could you figure out why?

File: tests/snuba/api/endpoints/test_organization_event_details.py (hunk 1 above)
Comment (markstory) [reply tests/snuba/api/endpoints/test_organization_event_details.py]: Right now there is no querying to the transactions dataset. I'm planning on adding that later. This is mostly a regression test for the future and to ensure that inserting into the transactions dataset doesn't cause an error.

**Comment by markstory** [comment]
@wedamija The extra time was all related to a timeout caused by invalid transaction event data being used in a fixture.

**Pull request merged**